dp-hello-world-event/service/interfaces.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-hello-world-event/config"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+)
+
+//go:generate moq -out mock/initialiser.go -pkg mock . Initialiser
+//go:generate moq -out mock/server.go -pkg mock . HTTPServer
+//go:generate moq -out mock/healthCheck.go -pkg mock . HealthChecker
+
+//go:generate moq -out mock/kafka.go -pkg mock . KafkaConsumer
+
+// Initialiser defines the methods to initialise external services
+type Initialiser interface {
```

Comment:
In filter API we took a different approach, instead of defining an initialiser with all the methods to initialise external dependencies, we simply created some functions in service.go to initialise dependencies (that would correspond to DoXYZ in initialiser), and then we can overwrite them for testing. 

For example:
- [get producer](https://github.com/ONSdigital/dp-filter-api/blob/develop/service/service.go#L60)
- [mock get producer](https://github.com/ONSdigital/dp-filter-api/blob/develop/service/service_test.go#L68)

In order to keep the state of external dependencies, instead of using `ExternalServiceList` struct, we simply keep the dependencies in the` Service` struct as pointers, and then check if the pointers are nil when needed.

For example:
- [close producer only if present](https://github.com/ONSdigital/dp-filter-api/blob/develop/service/service.go#L252)


I think the filter API approach is much clearer. Do you think it would be worth using the same approach for the generic service generation? (maybe it needs to be in a different PR?)

Reply:
I think there's another 20% task that wants to look at the service package and possibly moving stuff into a library. It might be better doing this as part of that task. This (and/or the api and controller examples) would then form the basis of how that library is used. That might be a better place to change the service and initialiser approach.

Reply:
It sounds good!